packages/clay-dropdown/src/ClayDropdownBase.js

```diff
@@ -2,17 +2,153 @@ import 'clay-button';
 import 'clay-checkbox';
 import 'clay-icon';
 import 'clay-radio';
-import Dropdown from 'metal-dropdown';
-import Soy from 'metal-soy';
+import {Align} from 'metal-position';
+import {core, object} from 'metal';
 import {Config} from 'metal-state';
-
+import Component from 'metal-component';
+import dom from 'metal-dom';
+import {EventHandler} from 'metal-events';
+import Soy from 'metal-soy';
 import itemsValidator from './items_validator';
 import templates from './ClayDropdownBase.soy.js';
 
 /**
  * Implementation of the base for Metal Clay Dropdown.
  */
-class ClayDropdownBase extends Dropdown {
+class ClayDropdownBase extends Component {
+	/**
+	 * @inheritDoc
+	 */
+	attached() {
+		super.attached();
+		this.eventHandler_.add(
+			dom.on(document, 'click', this.handleDocClick_.bind(this))
+		);
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	created() {
+		this.eventHandler_ = new EventHandler();
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	detached() {
+		super.attached();
+		this.eventHandler_.removeAllListeners();
+	}
+
+	/**
+	 * Closes the dropdown.
+	 * @protected
+	 */
+	close_() {
+		this.expanded = false;
+	}
+
+	/**
+	 * Handles document click in order to hide menu.
+	 * @param {!Event} event
+	 * @protected
+	 */
+	handleDocClick_(event) {
+		if (this.element.contains(event.target)) {
+			return;
+		}
+		this.close_();
+	}
+
+	/**
+	 * Toggles the dropdown, closing it when open or opening it when closed.
+	 */
+	toggle() {
+		this.expanded = !this.expanded;
+	}
+
+	/**
+	 * The setter function for the `classMap_` staet.
+	 * @param {Object} value
+	 * @return {!object}
+	 * @protected
+	 */
+	setterClassMapFn_(value) {
```

Comment:
classMap was used in metal-dropdown soy template. Since we're not using this param in our templates we can get rid of this method.